master/buildbot/db/logs.py

```diff
@@ -68,22 +69,27 @@ def thd(conn):
             q = q.order_by(tbl.c.first_line)
             rv = []
             for row in conn.execute(q):
-                assert not row.compressed, "compressed rows not supported yet"
-                content = row.content.decode('utf-8')
-                if row.first_line < first_line:
-                    idx = -1
-                    count = first_line - row.first_line
-                    for _ in xrange(count):
-                        idx = content.index('\n', idx + 1)
-                    content = content[idx + 1:]
-                if row.last_line > last_line:
-                    idx = len(content) + 1
-                    count = row.last_line - last_line
-                    for _ in xrange(count):
-                        idx = content.rindex('\n', 0, idx)
-                    content = content[:idx]
-                rv.append(content)
+                if row.compressed:
+                    # Uncompress the payload
+                    rv.append(zlib.decompress(row.content).decode('utf-8'))
+                else:
+                    # Extract all chunks
+                    content = row.content.decode('utf-8')
+                    if row.first_line < first_line:
```

Comment:
I think you need to untab this code.
Once the chunk is uncompressed, you still have to filter the lines requested.